### PR TITLE
Update commitment options

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Docs are out of date for SingleGossip
#### Summary of Changes
Update docs 
Fixes #

### CODE REVIEW

File: docs/src/apps/jsonrpc-api.md [hunk 1]
```diff
@@ -111,11 +111,14 @@ Requests can be sent in batches by sending an array of JSON-RPC request objects
 Solana nodes choose which bank state to query based on a commitment requirement
 set by the client. Clients may specify either:
 
-- `"max"` - the node will query the most recent bank confirmed by the cluster as having reached maximum lockout.
-- `"root"` - the node will query the most recent bank having reached maximum lockout on this node.
-- `"single"` - the node will query the most recent bank having reached 1 cluster confirmation.
-- `"singleGossip"` - the node will query the most recent bank having reached 1 cluster confirmation via gossip votes; may occur before or after `single`, depending on gossip traffic.
-- `"recent"` - the node will query its most recent bank.
+- `"max"` - the node will query the most recent bank confirmed by the cluster as having reached `MAX_LOCKOUT_HISTORY` confirmations
```
Comment (mvines) [inline docs/src/apps/jsonrpc-api.md:114]:
```suggestion
- `"max"` - the node will query the most recent bank confirmed by supermajority of the network as having reached `MAX_LOCKOUT_HISTORY` confirmations
```

Comment (mvines) [reply docs/src/apps/jsonrpc-api.md:114]: (to be consistent with `single`)

File: docs/src/apps/jsonrpc-api.md [hunk 2]
```diff
@@ -111,11 +111,14 @@ Requests can be sent in batches by sending an array of JSON-RPC request objects
...
+- `"root"` - the node will query the most recent bank having reached `MAX_LOCKOUT_HISTORY` confirmations on this node
+- `"single"` - the node will query the most recent bank having reached 1 confirmation by supermajority of the network.
```
Comment (mvines) [inline docs/src/apps/jsonrpc-api.md:116]: Re: use of "bank" here -- I feel we should use "block" instead as this information is more intended to be outward facing and we don't define "bank" at https://docs.solana.com/terminology

**Review by mvines — COMMENTED** [review]
Do we even have a use case for `single`?   Why not use `singleGossip` always?

File: docs/src/apps/jsonrpc-api.md [hunk 3]
```diff
@@ -111,11 +111,15 @@ Requests can be sent in batches by sending an array of JSON-RPC request objects
 Solana nodes choose which bank state to query based on a commitment requirement
 set by the client. Clients may specify either:
 
-- `"max"` - the node will query the most recent bank confirmed by the cluster as having reached maximum lockout.
-- `"root"` - the node will query the most recent bank having reached maximum lockout on this node.
-- `"single"` - the node will query the most recent bank having reached 1 cluster confirmation.
-- `"singleGossip"` - the node will query the most recent bank having reached 1 cluster confirmation via gossip votes; may occur before or after `single`, depending on gossip traffic.
-- `"recent"` - the node will query its most recent bank.
+- `"max"` - the node will query the most recent block confirmed by supermajority of the network as having reached
+maximum lockout on this node.
```
Comment (CriesofCarrots) [inline docs/src/apps/jsonrpc-api.md:115]:
```suggestion
maximum lockout.
```

File: docs/src/apps/jsonrpc-api.md [hunk 4]
```diff
@@ -111,11 +111,15 @@ Requests can be sent in batches by sending an array of JSON-RPC request objects
...
+- `"root"` - the node will query the most recent block having reached `MAX_LOCKOUT_HISTORY` confirmations on this node.
```
Comment (CriesofCarrots) [inline docs/src/apps/jsonrpc-api.md:116]:
```suggestion
- `"root"` - the node will query the most recent block having reached maximum lockout on this node.
```

**Review by CriesofCarrots — COMMENTED** [review]
Not to be pedantic, but don't we prefer `cluster` to `network`? Cluster is defined in terminology.md.

**Comment by carllin** [comment]
@mvines heh yeah, I think `Single` was a stopgap before `SingleGossip` was a thing, and now it's just for compatibility with downstream clients

**Review by CriesofCarrots — APPROVED** [review]

**Comment by mvines** [comment]
> @mvines heh yeah, I think `Single` was a stopgap before `SingleGossip` was a thing, and now it's just for compatibility with downstream clients

Oh let's yank it then!   Well for starters we can remove it from the JSON RPC documentation, and mark it as deprecated in `sdk/src/commitment_config.rs` as a comment

**Comment by mvines** [comment]
removed automerge since there's still instances of "network" in this PR, and let's remove "single" from the docs too while we're here

**Comment by carllin** [comment]
@mvines good catches! Updated as suggested.

File: docs/src/apps/jsonrpc-api.md [hunk 5]
```diff
@@ -1957,7 +1962,7 @@ The result is the latest vote, containing its hash, a list of voted slots, and a
 Unsubscribe from vote notifications
 
 #### Parameters:
-
+s
```
Comment (CriesofCarrots) [inline docs/src/apps/jsonrpc-api.md:1965]: Random "s"

**Review by mvines — DISMISSED** [review]

File: docs/src/apps/jsonrpc-api.md [hunk 6]
```diff
@@ -111,11 +111,16 @@ Requests can be sent in batches by sending an array of JSON-RPC request objects
...
-- `"single"` - the node will query the most recent bank having reached 1 cluster confirmation.
-- `"singleGossip"` - the node will query the most recent bank having reached 1 cluster confirmation via gossip votes; may occur before or after `single`, depending on gossip traffic.
-- `"recent"` - the node will query its most recent bank.
+- `"max"` - the node will query the most recent block confirmed by supermajority of the cluster as having reached
+maximum lockout.
+- `"root"` - the node will query the most recent block having reached maximum lockout on this node.
+- `"singleGossip"` - the node will query the most recent block that has been voted on by supermajority of the cluster via
+gossip votes. This differs from `single` in that:
+  - It incorporates votes from gossip and replay.
+  - It does not count votes on descendants of a bank, only direct votes on the bank.
```
Comment (mvines) [inline docs/src/apps/jsonrpc-api.md:120]:
```suggestion
  - It does not count votes on descendants of a block, only direct votes on that block.
```

File: sdk/src/commitment_config.rs [hunk 7]
```diff
@@ -62,11 +62,15 @@ pub enum CommitmentLevel {
...
-    /// The highest slot having reached 1 confirmation.
+    /// (DEPRECATED) The highest slot having reached 1 confirmation by supermajority of the cluster.
     Single,
 
-    /// The highest slot having reached 1 confirmation via gossip votes; may occur before or after Single,
-    /// depending on gossip traffic.
+    /// The highest slot that has been voted on by supermajority of the cluster
+    /// This differs from `single` in that:
+    /// 1) It incorporates votes from gossip and replay.
+    /// 2) It does not count votes on descendants of a bank, only direct votes on the bank.
```
Comment (mvines) [inline sdk/src/commitment_config.rs:71]:
```suggestion
    /// 2) It does not count votes on descendants of a block, only direct votes on that block.
```

File: docs/src/apps/jsonrpc-api.md [hunk 8]
```diff
@@ -111,11 +111,16 @@ Requests can be sent in batches by sending an array of JSON-RPC request objects
...
-- `"max"` - the node will query the most recent bank confirmed by the cluster as having reached maximum lockout.
-- `"root"` - the node will query the most recent bank having reached maximum lockout on this node.
-- `"single"` - the node will query the most recent bank having reached 1 cluster confirmation.
-- `"singleGossip"` - the node will query the most recent bank having reached 1 cluster confirmation via gossip votes; may occur before or after `single`, depending on gossip traffic.
-- `"recent"` - the node will query its most recent bank.
+- `"max"` - the node will query the most recent block confirmed by supermajority of the cluster as having reached
+maximum lockout.
+- `"root"` - the node will query the most recent block having reached maximum lockout on this node.
+- `"singleGossip"` - the node will query the most recent block that has been voted on by supermajority of the cluster via
+gossip votes. This differs from `single` in that:
```
Comment (mvines) [inline docs/src/apps/jsonrpc-api.md:118]:
```suggestion
gossip votes.
```
(single is no longer in here, so we can't reference it)

**Pull request merged**